python/ct/proto/client_v2.proto

```diff
@@ -0,0 +1,180 @@
+syntax = "proto2";
+
+package ct;
+
+import "ct/proto/client.proto";
+import "ct/proto/tls_options.proto";
+
+// Actually X509ChainEntry in RFC6962-bis
+message X509ChainEntryV2 {
+  // leaf_certificate, opaque (ASN.1Cert) <1..2^24-1>
+  optional bytes leaf_certificate = 1 [ (tls_opts).min_length = 1,
+                                        (tls_opts).max_length = 0xffffff ];
+  // certificate_chain, repeated opaque (ASN.1Cert) <0..2^24-1>
+  // A chain from the leaf to a trusted root, excluding the leaf, including
+  // the trusted root. May be empty if the submitted leaf_certificate is a
+  // trusted root.
+  repeated bytes certificate_chain = 2 [ (tls_opts).min_length = 0,
+                                         (tls_opts).max_length = 0xffffff,
+                                         (tls_opts).max_total_length = 0xffffff ];
+}
+
+message PrecertChainEntryV2 {
+  // pre_certificate, opaque (CMSPrecert) <1..2^24-1>
+  optional bytes pre_certificate = 1 [
+    (tls_opts).min_length = 1,
+    (tls_opts).max_length = 0xffffff ];
+
+  // precertificate_chain, repeated opaque (ASN.1Cert) <1..2^24-1>
+  // A chain from the precert to a trusted root, excluding the precert itself.
+  // Must have at least one certificate.
+  repeated bytes precertificate_chain = 2 [
+    (tls_opts).min_length = 0,
+    (tls_opts).max_length = 0xffffff,
+    (tls_opts).min_total_length = 1,
+    (tls_opts).max_total_length = 0xffffff ];
+}
+
+message LogIDV2 {
+  // DER-encoded OID without the tag byte.
+  optional bytes log_id = 1 [ (tls_opts).min_length = 2,
+                              (tls_opts).max_length = 0x7f];
+}
+
+enum SctExtensionType {
+  option (tls_enum_opts).max_value = 65535;
+
+  SCT_EXTENSION_RESERVED=65535;
+}
+
+message SctExtensionV2 {
+  optional SctExtensionType type = 1;
```

Comment:
`sct_extension_type` would be more consistent with the field below and with the draft.